BigBaseV2/src/services/gta_data/gta_data_service.cpp

```diff
@@ -85,27 +93,61 @@ namespace big
 		}
 	}
 
-	std::vector<std::string>& gta_data_service::get_ped_type_arr()
+	const std::vector<std::string>& gta_data_service::get_ped_type_arr()
 	{
 		return m_ped_type_arr;
 	}
 
-	std::vector<ped_item>& gta_data_service::get_ped_arr()
+	const std::vector<ped_item>& gta_data_service::get_ped_arr()
 	{
 		return m_ped_item_arr;
 	}
 
 
+	const weapon_item& gta_data_service::find_weapon_by_hash(Hash hash)
+	{
+		int idx = -1;
+
+		if (m_weapon_hash_idx_map.count(hash))
+		{
+			idx = m_weapon_hash_idx_map[hash];
+		}
 
-	void gta_data_service::load_from_file(file file_to_load, file file_etag, std::string url, void(gta_data_service::* load_func)(), std::string data_name)
+		if (idx == -1)
+		{
+			return empty_weapon_item;
+		}
+		else
+		{
+			return m_weapon_item_arr[idx];
+		}
```

Comment:
redundant else as you've already returned (exit) out of the function in the if statement.